Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleTableView.cpp

```diff
@@ -99,36 +160,89 @@ namespace ShaderManagementConsole
             m_shaderOptionDescriptors.push_back(shaderOptionDescriptor);
         }
 
-        // Sort all of the descriptors by name so that the columns are arranged in alphabetical order
-        AZStd::sort(
-            m_shaderOptionDescriptors.begin(),
-            m_shaderOptionDescriptors.end(),
-            [](const auto& a, const auto& b)
+        switch (m_columnSortMode)
+        {
+        case Alpha:
             {
-                return a.GetName().GetStringView() < b.GetName().GetStringView();
-            });
+                // Sort descriptors by name
+                AZStd::sort(
+                    m_shaderOptionDescriptors.begin(),
+                    m_shaderOptionDescriptors.end(),
+                    [](const auto& a, const auto& b)
+                    {   //                      small first ('a' first)
+                        return a.GetName().GetStringView() < b.GetName().GetStringView();
+                    });
+            }
+            break;
+        case Rank:
+            {
+                // Sort by shader declaration order
+                AZStd::sort(
+                    m_shaderOptionDescriptors.begin(),
+                    m_shaderOptionDescriptors.end(),
+                    [](const auto& a, const auto& b)
+                    {   //  small first (order 0 is highest)
+                        return a.GetOrder() < b.GetOrder();
+                    });
+            }
+            break;
+        case Cost:
+            {
+                // Sort by cost (high score first)
+                AZStd::sort(
+                    m_shaderOptionDescriptors.begin(),
+                    m_shaderOptionDescriptors.end(),
+                    [](const auto& a, const auto& b)
+                    {   //      big first (high score is more important)
+                        return a.GetCostEstimate() > b.GetCostEstimate();
+                    });
+            }
+            break;
+        }
```

Comment:
If someone sees a way to factorize this better, maybe through a lambda inside the lambda?